### PR TITLE
places arrow in beginning of first run

### LINKED ISSUE #232: Ensure shortcut arrow has all the right placement & logic
This is a stub to ensure we're checking how the shortcut arrow on the "welcome" page of the core add-on actually points to the right spot. Additionally, the shortcut arrow shouldn't appear in all cases.

Sample image:
add-on_ welcome to Rally

### PR DESCRIPTION
**Pull request by hamilton**
closes #232 

Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

File: core-addon/Storage.js [hunk 1]
```diff
@@ -64,6 +64,27 @@ module.exports = class Storage {
...
+    return await this.getItem("firstRunCompleted");
+  }
+
+  /**
+   * Set the provided demographic data.
+   *
+   * @returns {Promise} 
+   */
+  async setFirstRunCompletion(firstRun) {
+    console.log('FIRST RUN COMPLETION SET', firstRun)
```
Comment (rhelmer) [inline core-addon/Storage.js]: Is this just for debugging or did you intend to leave this @hamilton ?

File: src/App.svelte [hunk 2]
```diff
@@ -32,7 +33,14 @@
       <!-- onboarding flow -->
       <!-- the onboarding-complete event occurs once the user has
       gotten through the profile completion step. -->
+      <!-- the first-run-complete event occurs once the user has
+      clicked on "Get Started" on the Welcome page. -->
       <Onboarding
+        firstRunCompleted={$store.firstRunCompleted}
+        on:first-run-initiated={() => {
+          console.log('did this fire?')
```
Comment (rhelmer) [inline src/App.svelte]: Same question re: debugging :)

File: src/routes/welcome/Content.svelte [hunk 3]
```diff
@@ -104,3 +111,6 @@
     </div>
   </div>
 </section>
+{#if showArrow}
+  <FirstRunArrow />
+{/if}
```
Comment (rhelmer) [inline src/routes/welcome/Content.svelte:116]: Could you fix the newline please?

File: src/web-extension-api.js [hunk 4]
```diff
@@ -202,6 +212,7 @@ export default {
     switch (message.type) {
       case "update-state": {
         // update the UI.
+        console.log("updatestate")
```
Comment (rhelmer) [inline src/web-extension-api.js:215]: Moar debug-looking logging

**Review by rhelmer — APPROVED** [review]
lgtm overall, just some console logging you should remove and a newline to add, please do this before merging.

File: src/web-extension-api.js [hunk 5]
```diff
@@ -186,6 +187,15 @@ export default {
     await sendToCore(this._connectionPort, "update-demographics", data);
   },
 
+  /**
+   * Updates the stored version of the demographics data.
```
Comment (Dexterp37) [inline src/web-extension-api.js:191]: nit: the comment is wrong

File: core-addon/Core.js [hunk 6]
```diff
@@ -558,13 +561,15 @@ module.exports = class Core {
    */
   async _sendStateUpdateToUI() {
     let enrolled = !!(await this._storage.getRallyID());
```
Comment (Dexterp37) [inline core-addon/Core.js:563]: We already only set the `enrolled` status if after user goes through the onboarding and consents to join rally. Isn't this enough?

Comment (hamilton) [reply core-addon/Core.js:563]: It isn't. We only want the arrow to appear on the very first load of the welcome page.

**Pull request merged**